src/worker.rs

```diff
@@ -69,13 +69,13 @@ impl<T: Socket + ?Sized> Worker<T> {
 
     /// Sends a file to the remote [`SocketAddr`] that has sent a read request using
     /// a random port, asynchronously.
-    pub fn send(self) -> Result<(), Box<dyn Error>> {
+    pub fn send(self, rcv_oack_ack: bool) -> Result<(), Box<dyn Error>> {
```

Comment:
small nitpick: can it be `check_response` or `should_check_response` instead of `rcv_oack_ack`?